### PR TITLE
Refactor replacing multiple Seqs for argument attribures with a single Seq

### PR DESCRIPTION
**Pull request by russellremple**
Changes rawArgs: Seq[String], mappedArgs: Seq[String], argTypes: Seq[Type], hasDefaults: Seq[Boolean], forceDefaultNone: Seq[Boolean], omitDefaults: Seq[Boolean], and default: Seq[c.Tree] into a single Seq[Argument] that captures all the attributes for an argument.

I recognize this may make merging with upickle harder in the future, but I think it is a cleaner setup for handling arguments, and will help us with future extensions.

@htmldoug @jducoeur 

P.S. Includes all the merges with https://github.com/rallyhealth/weePickle/pull/33

### CODE REVIEW

**Comment by htmldoug** [comment]
> this may make merging with upickle harder in the future

...unless you submit a PR upstream!

File: implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala [hunk 1]
```diff
@@ -51,32 +47,96 @@ object Macros {
...
-      val defaults =
-        for(((hasDefault,forceDefault), i) <- hasDefaults.zip(forceDefaultNone).zipWithIndex)
-        yield {
-          val defaultName = TermName("apply$default$" + (i + 1))
-          if (!hasDefault) {
-            if (forceDefault) q"${TermName("None")}"
-            else q"null"
-          }
-          else q"$companion.$defaultName"
+    case class Argument(
```
Comment (htmldoug) [inline implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala]: Probably a good idea to make this private so we can evolve it without breaking bin compat.

Comment (russellremple) [reply implicits/src/com/rallyhealth/weepickle/v1/implicits/internal/Macros.scala]: Package private ([internal]) was the best I could do.

**Comment by russellremple** [comment]
Merge please? (I still don't have merge priv.)

**Pull request merged**